tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Proxy/InferredProxyCoordinator.cs

```diff
@@ -0,0 +1,75 @@
+// <copyright file="InferredProxyCoordinator.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Propagators;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Proxy;
+
+/// <summary>
+/// Coordinates the extraction of proxy metadata and creation of proxy spans.
+/// Acts as the main entry point for creating inferred proxy spans.
+/// </summary>
+internal class InferredProxyCoordinator
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<InferredProxyCoordinator>();
+    private readonly IInferredSpanFactory _spanFactory;
+    private readonly IInferredProxyExtractor _extractor;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="InferredProxyCoordinator"/> class.
+    /// </summary>
+    /// <param name="extractor">Component that extracts proxy metadata from headers</param>
+    /// <param name="spanFactory">Component that creates spans from the extracted metadata</param>
+    public InferredProxyCoordinator(IInferredProxyExtractor extractor, IInferredSpanFactory spanFactory)
+    {
+        _spanFactory = spanFactory;
+        _extractor = extractor;
+    }
+
+    /// <summary>
+    /// Extracts proxy metadata from headers and creates a corresponding span.
+    /// </summary>
+    /// <returns>
+    /// When successful, returns an object containing:
+    /// <list type="bullet">
+    /// <item>The created <see cref="Scope"/> containing the proxy <see cref="Span"/></item>
+    /// <item>An updated <see cref="PropagationContext"/> that includes the new <see cref="SpanContext"/> and previous <see cref="Baggage"/>.</item>
+    /// </list>
+    /// Returns <see langword="null"/> if extraction fails or span creation fails.
+    /// </returns>
+    public InferredProxyScopePropagationContext? ExtractAndCreateScope<TCarrier, TCarrierGetter>(
+        Tracer tracer,
+        TCarrier carrier,
+        TCarrierGetter carrierGetter,
+        PropagationContext propagationContext)
+        where TCarrierGetter : struct, ICarrierGetter<TCarrier>
+    {
+        try
+        {
+            if (!_extractor.TryExtract(carrier, carrierGetter, tracer, out var proxyData))
+            {
+                return null;
+            }
+
+            var scope = _spanFactory.CreateSpan(tracer, proxyData, propagationContext.SpanContext);
+            if (scope == null)
+            {
+                return null;
+            }
+
+            var updatedContext = new PropagationContext(scope.Span.Context, propagationContext.Baggage); // TODO is this correct?
```

Comment:
Just wondering about this TODO, is this answered or a possible issue?

Reply:
I think that this is _okay_ it seems to work, it isn't a well phrased TODO, I was mainly asking myself if the `propagationContext.Baggage` was correct or if I needed to create new baggage and merge the old into it.

Reply:
LGTM, removed the TODO comment.